src/containers/layouts/messaging/chat2/chat2.component.tsx

```diff
@@ -0,0 +1,222 @@
+import React from 'react';
+import {
+  Dimensions,
+  FlatList,
+  ListRenderItemInfo,
+  View,
+  ViewProps,
+} from 'react-native';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import {
+  List,
+  Input,
+  Button,
+  ButtonProps,
+} from '@kitten/ui';
+import {
+  Conversation as ConversationModel,
+  Message as MessageModel,
+} from '@src/core/model';
+import {
+  ChatMessage,
+  ChatMessageProps,
+  ChatMessageAlignment,
+  ChatFileMessageProps,
+  ChatFileMessage,
+  ChatFileMessageAppearance,
+} from '@src/components/messaging';
+import {
+  profile1,
+  profile2,
+} from '@src/core/data/profile';
+import {
+  PlusIcon,
+  MicIcon,
+  PaperPlaneIcon,
+} from '@src/assets/icons';
+import { UiMessageModel } from '../uiMessage.model';
+import { Chat2FileSection } from './chat2FileSection.component';
+import { MediaLibrary } from 'expo';
+
+const width: number = Dimensions.get('window').width;
+
+interface ComponentProps {
+  conversation: ConversationModel;
+  fileSectionOpened: boolean;
+  newMessage: string;
+  galleryFiles: MediaLibrary.Asset[];
+  fileMessageAppearance: ChatFileMessageAppearance; // just for demo purposes
+  onNewMessageChange: (text: string) => void;
+  onMessageAdd: () => void;
+  onAddButtonPress: () => void;
+  onCancelButtonPress: () => void;
+  onPhotoOrVideoButtonPress: () => void;
+  onFileButtonPress: () => void;
+  onLocationButtonPress: () => void;
+  onContactButtonPress: () => void;
+  onTakePhotoButtonPress: () => void;
+  onGalleryItemPress: (item: MediaLibrary.Asset) => void;
+}
+
+export type Chat2ComponentProps = ThemedComponentProps & ComponentProps;
+
+class Chat2Component extends React.Component<Chat2ComponentProps> {
+
+  private listRef: React.RefObject<FlatList<any>> = React.createRef();
+
+  private onNewMessageChange = (text: string): void => {
+    this.props.onNewMessageChange(text);
+  };
+
+  private onMessageAdd = (): void => {
+    this.props.onMessageAdd();
+    this.listRef.current.scrollToEnd();
+  };
+
+  private onAddButtonPress = (): void => {
+    this.props.onAddButtonPress();
+  };
+
+  private createUiMessages = (): UiMessageModel[] => {
```

Comment:
oh, that's really nice
so if your model is UI model, why really not using enums to provide views?